### PR TITLE
#185106781 view and update user profile

### PR DESCRIPTION
**Pull request by rwamugema**
####What does this pr do?

- [ ] View user profile.
- [ ] Update user profile.

### How should this be manually tested?
Installation
1. Clone the repository by running

git clone https://github.com/atlp-rwanda/falcons-ec-fe.git

**3. Install the necessary dependencies by running **

`npm install`

6. Run react development server by running

`npm run dev`

### Some background context?

Go to this route http://localhost:3000/profile

####Screenshots?
Screenshot 2023-05-18 at 15 54 41
Screenshot 2023-05-22 at 15 49 12
Screenshot 2023-05-22 at 15 50 29

Questions:
none

### CODE REVIEW

File: src/hooks/useFormData.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable react/prop-types */
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import updateProfile from '../redux/slices/profile/updateProfile';
```
Comment (fab-ryan) [inline src/hooks/useFormData.js]: this is not a hook if you wish to call or create a hook don't specify data that need to be sent there

File: src/main.jsx [hunk 2]
```diff
@@ -1,14 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { Provider } from 'react-redux';
+import { ToastContainer } from 'react-toastify';
 import App from './App.jsx';
 import './styles/index.css';
+import './styles/profile.css';
```
Comment (fab-ryan) [inline src/main.jsx]: add this line in profile file

File: src/redux/slices/profile/updateProfile.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            state.error = null;
+        })
+        .addCase(updateProfile.rejected, (state, {payload})=>{
+            state.isLoading = false;
+            state.profileData = null;
+            state.error = payload;
+        })
+    }
+ })
+ export const updateProfileActions = updateProfileSlices.actions;
```
Comment (fab-ryan) [inline src/redux/slices/profile/updateProfile.js]: combine all action from this file

File: src/redux/slices/profile/getProfileSlice.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const initialState = {
+    profile: null,
+    isLoading: false,
+    error: null,
+}
+
+const getUser = createAsyncThunk(
+    'user/getProfile', async () => {
+        const token = localStorage.getItem('token');
+        const profile = await fetch(`https://e-commerce-falcons.onrender.com/api/v1/users/profile/single`,{
```
Comment (fab-ryan) [inline src/redux/slices/profile/getProfileSlice.js]: please combine getProfileSlice and updateProfile to single file like: profileSlice

File: src/redux/store.js [hunk 5]
```diff
@@ -1,24 +1,25 @@
...
 import categoriesSlice from './slices/product/categories';
 import signupSlice from './slices/user/signup';
 
 const reducers = {
-  counter: counterReducer,
-};
+  profile: getProfileReducer,
+  profileUpdate: updateProfileReducer
+}
 const store = configureStore({
```
Comment (fab-ryan) [inline src/redux/store.js]: please follow well other's format don't create you own you are confusing the team

File: src/utils/getFormData.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const getFormFromObject = (data) => {
+    const formData = new FormData();
+    if (!data) {
+      return null;
+    }
+    Object.keys(data).forEach((eachKey) => {
+      if (Array.isArray(data[eachKey])) {
+        data[eachKey].forEach((each) => formData.append(eachKey, each));
```
Comment (fab-ryan) [inline src/utils/getFormData.js]: delete that  hook and use this or make this a hook

**Review by fab-ryan — COMMENTED** [review]
work on those comments

**Review by fab-ryan — APPROVED** [review]

**Review by masterchief-00 — APPROVED** [review]

**Review by kirengaboris — APPROVED** [review]

**Pull request merged**